heat/utils/data/matrixgallery.py

```diff
@@ -1,153 +1,186 @@
-    m: int,
-    n: int,
-    r: int,
-    quantile_function: Callable = lambda x: -log(x),
-    split: Union[None, int] = None,
-    device: Union[None, str, Device] = None,
-    comm: Union[None, Communication] = None,
-    dtype: Type[datatype] = core.float32,
-) -> Tuple[DNDarray, Tuple[DNDarray]]:
-    """
-    Creates a random m x n matrix with rank r.
-    This routine uses :func:`random_known_singularvalues` with r singular values randomly chosen
-    w.r.t. the distribution with quantile function given by the input quantile_function. Default yields exponential distibution with parameter lambda=1.
-    Unlike in :func:`random_known_singularvalues`, here the singular values of the output are sorted in descending order.
-    """
-    if r > min(m, n):
-        raise RuntimeError("rank must not exceed matrix dimensions.")
-
-    singular_values = rand(r, dtype=dtype, comm=comm, device=device)
-    singular_values = sort(quantile_function(singular_values), descending=True)[0]
-
-    return random_known_singularvalues(
-        m, n, singular_values, split=split, device=device, comm=comm, dtype=dtype
-    )
+"""
+Generate matrices for specific tests and functions
+"""
+
+from heat import core
+from ...core.dndarray import DNDarray
+from ...core.communication import Communication
+from ...core.devices import Device
+from ...core.types import datatype
+from ...core.random import randn, rand
+from ...core.linalg import qr, matmul
+from ...core.manipulations import diag, sort
+from ...core.exponential import log
+from typing import Type, Union, Tuple, Callable
+
+__all__ = ["hermitian", "parter", "random_known_singularvalues", "random_known_rank"]
+
+
+def hermitian(
+    n: int,
+    split: Union[None, int] = None,
+    device: Union[None, str, Device] = None,
+    comm: Union[None, Communication] = None,
+    dtype: Type[datatype] = core.complex64,
+) -> DNDarray:
+    """
+    Generates a Hermitian matrix of size `(n,n)`. A Hermitian matrix is a  complex square matrix that is equal to its conjugate transpose.
+
+    Parameters
+    ----------
+    n : int
+        size of the resulting square matrix
+    split: None or int, optional
+        The axis along which the array content is split and distributed in memory.
+    device: None or str or Device, optional
+        Specifies the device the tensor shall be allocated on, defaults globally set default device.
+    comm : Communication, optional
+        Handle to the nodes holding distributed parts or copies of this array.
+    dtype: Type[datatype], optional
+        The desired data-type for the array, defaults to ht.complex64.
+
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Hermitian_matrix
+    """
+    real_dtype = core.float32 if dtype is core.complex64 else core.float64
+    matrix = randn(n, n, dtype=real_dtype, split=split, device=device)
+    matrix += 1j * randn(n, n, dtype=real_dtype, split=split, device=device)
+    return matrix @ core.conj(matrix).T
```

Comment:
Two comments: 
* if we scale by 1/matrix.shape[0] the singular values are distributed according to Marchenko Pastur distribution (see above)... also this scaling might remove unwanted effects when choosing larger and larger matrices 
* The formula yields, of course, and they are even positive definite (with probability 1? - Im not an expert in this stuff...). However, doing the matrix-multiplication is quite expensive ... can we add something like an option "positive_definite" that triggers the formula used so far and use 
```
return matrix + core.conj(matrix).T.resplit(matrix.split) 
```
instead for the remaining cases?

Reply:
Thanks, done